python/ray/air/_internal/torch_utils.py

```diff
@@ -119,6 +119,9 @@ def convert_ndarray_to_torch_tensor(
     Returns: A Torch Tensor.
     """
     ndarray = _unwrap_ndarray_object_type_if_needed(ndarray)
+    # NOTE: PyTorch raises a `UserWarning` if `ndarray` isn't writeable. See #28003.
+    if not ndarray.flags["WRITEABLE"]:
+        ndarray = np.copy(ndarray)
     return torch.as_tensor(ndarray, dtype=dtype, device=device)
```

Comment:
I don't think that copying the ndarray here is the right solution, since we _want_ to be able to reuse the shared memory tensor data buffers without incurring unnecessary copies. We're also under the expectation the user/model will not mutate these input tensors during training and inference, so I don't think that the copy is necessary for correctness.

What if we suppressed the warning instead via something like:
```suggestion
    with warnings.catch_warnings():
        warnings.simplefilter("ignore")
        tensor = torch.as_tensor(ndarray, dtype=dtype, device=device)
    return tensor
```

Reply:
Are we confident that the warning is harmless? If so, I think this is fine/

Reply:
If the user tries to mutate the tensor (e.g. with in-place tensor operations) then this will be undefined behavior, but the common case is that this tensor data buffer will be left untouched in Plasma until we either (a) transfer the tensor to the GPU, or (b) we create a new tensor via an operation that creates a copy; in either case, we're not likely to mutate it.

Reply:
I'm not sure if we should suppress the warning, though, since it's a useful signal to the user that they will need to `.clone()` it if they wish to mutate it.

Reply:
It'd be great to suppress the *repeated* warnings (if that is somehow possible).